Airbnb.Infrastructure/Repositories/UnitOfWork.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Airbnb.Domain.Entities;
+using Airbnb.Domain.Interfaces.Repositories;
+using Airbnb.Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace Airbnb.Infrastructure.Repositories
+{
+    public class UnitOfWork: IUnitOfWork
+    {
+        private readonly AirbnbDbContext _context;
+        private IGenericRepository<Booking, int> _bookingRepository;
+        private IGenericRepository<Category, int> _categoryRepository;
+        private IGenericRepository<Country, int> _countryRepository;
+        private IGenericRepository<Image, int> _imageRepository;
+        private IGenericRepository<Location, int> _locationRepository;
+        private IGenericRepository<Property, int> _propertyRepository;
+        private IGenericRepository<PropertyCategory, int> _propertyCategoryRepository;
+        private IGenericRepository<Review, int> _reviewRepository;
```

Comment:
That was a great job, but I have an enhancement idea: what if we made it more generic instead of adding each one individually? You could use a hashtable to store them. If an item already exists, you wouldn't need to create a new one; otherwise, it would still be stored.